include/subgroup/tile/impl/load_xe.hpp

```diff
@@ -531,6 +533,77 @@ tile_load(tile_t& tile, payload_t& payload) {
   }
 }
 
+/// @brief This function loads data from unaligned-2D memory surface.
+/// Loads an array of rectangular regions (X,Y)..(X+W,Y+H) from memory into
+/// registers. Each block will be loaded serially by its corresponding payload.
+/// @tparam tile_t Is the tile_t struct contains registers.
+/// These registers will be the destination of load operation.
+/// @tparam payload_t Is the mem_payload_t struct describing the memory
+/// information. Payload indicates the source of load operation.
+/// @tparam L1 Is the cache hint for L1 cache.
+/// @tparam L3 Is the cache hint for L3 cache.
+/// @param tile Is the tile object with type tile_t, holds the return data of
+/// the loads.
+/// @param payload Is the payload object with type payload_t. Contains all the
+/// information for loads.
+/// @return No return, update in place.
+template <
+    cache_hint L1 = cache_hint::cached,
+    cache_hint L3 = cache_hint::cached,
+    typename tile_t,
+    typename payload_t>
+__XETLA_API typename std::enable_if_t<
+    detail::check_load_type<tile_t, payload_t>::is_global_block_2d &&
+    !detail::check_load_type<tile_t, payload_t>::is_lsc_gather &&
+    payload_t::arch_tag <= gpu_arch::XeHpg>
```

Comment:
how about align to https://github.com/intel-innersource/frameworks.ai.pytorch.ipex-gpu/pull/4100

```suggestion
    !arch_has_2d_load_store(payload_t::arch_tag)
```